### DISCUSSION TITLE
Is there any way to use TopicFilter in user code?

### DISCUSSION THREAD

**Discussion by Antiarchitect** [ntex-rs/ntex-mqtt#115] [Q&A]
I'd like to use TopicFilter and associated functions like `is_valid` and `matches_topic` in order to match my Subscriptions filter against incoming events topics. Could this module be public or there is another way?

**Comment by fafhrd91** [comment]
matches_topic/filter methods are public. i am not sure about is_valid, you can create topic filter with TryFrom<ByteString> and FromStr traits

**Comment by Antiarchitect** [comment]
I see they're marked as public, but I don't know how to use them properly. Now I have:

```
use ntex_mqtt::topic::TopicFilter;

TopicFilter::from(self.sub).matches_topic(topic)
```

and got:

```
error[E0603]: module `topic` is private
 --> src/mqtt/sessions/subscription.rs:2:17
  |
2 | use ntex_mqtt::{topic::TopicFilter, types::QoS};
  |                 ^^^^^ private module
```

**Comment by fafhrd91** [comment]
use ntex_mqtt::TopicFilter

**Comment by Antiarchitect** [comment]
Thank you @fafhrd91
